sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ChangeFeedProcessorContext.java

```diff
@@ -25,4 +25,14 @@ public interface ChangeFeedProcessorContext {
      * */
     @Beta(value = Beta.SinceVersion.V4_51_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
     String getLeaseToken();
+
+    /**
+     * Get the diagnostics from the underlying feed response.
+     *
+     * @return The diagnostics object.
+     */
+    @Beta(value = Beta.SinceVersion.V4_64_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    default CosmosDiagnostics getDiagnostics() {
```

Comment:
Return type should be CosmosDiagnosticsContext - a single batch of items provided here could have resulted in multiple calls to the backend (hedging / availability strategy) - so we should provide the CosmsoDiagnosticsContext.

The reason why other API doesn't do this is simply that the API existed before CosmsoDiagnosticsContext was added. New APs should stick to CosmosDiagnosticsContext

Reply:
Updated